src/Bishop.cpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "Bishop.h"
+
+/**
+ * Checks if the move is valid for a Bishop piece.
+ *  
+ * @param moveCoords A pair of pairs of integers representing the start and end coordinates of the move.
+ * @param chessMatrix A 2D vector of unique_ptr to Piece representing the current state of the chess board.
+ * @return true if the move is valid for the Bishop piece, false otherwise.
+ */
+const bool Bishop::isValidMove(const std::pair<std::pair<int, int>, std::pair<int, int>>& moveCoords, const std::vector<std::vector<std::unique_ptr<Piece>>>& chessMatrix)
```

Comment:
Instead of embedding the logic for checking if the path is clear and if the destination square is occupied within the isValidMove function, consider extracting these checks into separate helper functions. This promotes code modularity and improves readability.

Reply:
Done.